src/coreclr/jit/llvm.cpp

```diff
@@ -2392,6 +2451,64 @@ void Llvm::PlaceAndConvertShadowStackLocals()
     lowerToShadowStack();
 }
 
+static bool placeLocalInAlloca(LclVarDsc* varDsc)
+{
+    return !varDsc->lvTracked || varDsc->lvHasLocalAddr;
+}
```

Comment:
I suppose with `DontTrackLocalsWithAddress` this becomes just `!varDsc->lvTracked`.